### PR TITLE
ARTEMIS-3075 Skip temporary queues scale down

### PR DESCRIPTION
**Pull request by brusdev**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ScaleDownHandler.java [hunk 1]
```diff
@@ -124,9 +124,11 @@ public long scaleDownMessages(ClientSessionFactory sessionFactory,
                for (Binding binding : bindings.getBindings()) {
                   if (binding instanceof LocalQueueBinding) {
                      Queue queue = ((LocalQueueBinding) binding).getQueue();
-                     // as part of scale down we will cancel any scheduled message and pass it to theWhile we scan for the queues we will also cancel any scheduled messages and deliver them right away
-                     queue.deliverScheduledMessages();
-                     queues.add(queue);
+                     if (!queue.isTemporary()) {
+                        // as part of scale down we will cancel any scheduled message and pass it to theWhile we scan for the queues we will also cancel any scheduled messages and deliver them right away
+                        queue.deliverScheduledMessages();
```
[the remark this answers is not in the document]
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ScaleDownHandler.java:129]: Ohhh.. never mind... queues.add


I thought there was another loop looking for the search and I missed it.. ignore me.

**Comment by jbertram** [comment]
One of the benefits of scale-down as opposed to normal fail-over is that non-persistent messages are preserved. In normal fail-over only persistent messages are preserved. 

It is, of course, normal for temporary queues to disappear when a client disconnects, but it also might be nice for scale-down to preserve these temporary queues so clients can reconnect and continue their work since the client itself was not responsible for the disconnect. For normal clients using temporary queues who get disconnect because some kind of spurious network issue they are able to reconnect back to their temporary queue unless the connection ttl has elapsed and the broker has cleaned up the queue. This is discussed in [the documentation](http://activemq.apache.org/components/artemis/documentation/latest/client-reconnection.html).

**Comment by brusdev** [comment]
@jbertram I see your point my only concern is that the scale-down transforms temporary queues in not temporary queues. If no client will consume those queues after the scale-down they will never be cleaned up.

**Comment by jbertram** [comment]
@brusdev, that's definitely a problem. If we decide to support temporary queues we would need to fix that.

I also just realized that any temporary queue created during scale-down would not be directly tied to a client connection which means we'd have to clean them up some other way. We already have `org.apache.activemq.artemis.core.postoffice.impl.PostOfficeImpl.AddressQueueReaper` which could probably do the work, but it might be kind of fragile.

At this point it's probably best to just skip the temp queues as you've done. Keeping them would be nice, but it would almost certainly be a mess to implement.

**Pull request closed without merging**